packages/utils/hooks/useDebounce/index.ts

```diff
@@ -1,11 +1,11 @@
 'use client'
 
-import { useEffect, useRef } from 'react'
+import { useEffect, useState, useRef } from 'react'
 
 import debounce from 'lodash/debounce'
 
-const useDebounce = <T extends (params: any) => any>(
-  functionToDebounce: T,
+const useDebounce = <T extends any[], U>(
+  functionToDebounce: (...params: T) => U,
```

Comment:
I don't understand why we would need to manually define a return type here, could you explain?

Reply:
I guess this could have been 
```
const useDebounce = <T extends (...params: any[]) => any>(
  functionToDebounce: T,
  { debounceTime = 400 } = {},
) => ...
```
I just added it as it's how I'd personally write this generic, but I think it's the same thing functionally.